packages/babel-compiler/babel-compiler.js

```diff
@@ -160,8 +160,16 @@ BCp._inferFromBabelRc = function (inputFile, babelOptions, cacheDeps) {
   var babelrcPath = inputFile.findControlFile(".babelrc");
   if (babelrcPath) {
     if (! hasOwn.call(this._babelrcCache, babelrcPath)) {
-      this._babelrcCache[babelrcPath] =
-        JSON.parse(inputFile.readAndWatchFile(babelrcPath));
+      try {
+        this._babelrcCache[babelrcPath] =
+          JSON.parse(inputFile.readAndWatchFile(babelrcPath));
+      } catch (e) {
+        if (e instanceof SyntaxError) {
+          throw new Error(".babelrc is not a valid JSON file: " + e.message);
```

Comment:
Seems like this gets rid of the stacktrace in `e` as it's not passed into the new error being thrown.

Reply:
Are you sure? The stack trace should be similar because the error is thrown at (almost) the same source location as before.

Reply:
I'm thinking it's loosing the stack trace from the command that threw the error we're catching.

Reply:
What about just `e.message = ".babelrc is not a valid JSON file: " + e.message`?

Reply:
Sounds good!